airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from contextlib import contextmanager
+from io import IOBase
+from typing import Any, Dict, Iterable, List, Optional, Set
+
+import boto3.session
+import smart_open
+from botocore.client import BaseClient, Config
+from airbyte_cdk.sources.file_based.exceptions import ErrorListingFiles, FileBasedSourceError
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader
+from airbyte_cdk.sources.file_based.remote_file import FileReadMode, RemoteFile
+from source_s3.v4.config import Config
+
+
+class SourceS3StreamReader(AbstractFileBasedStreamReader):
+
+    def __init__(self):
+        super().__init__()
+        self._s3_client = None
+
+    @property
+    def config(self) -> Config:
+        return self._config
+
+    @config.setter
+    def config(self, value: Config):
```

Comment:
Curious, does mypy checks not flag this since the interface method expects `AbstractFileBasedSpec` even though `Config` is a subclass of it